core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1446,6 +1446,14 @@ public long getLastWriteNanos() {
     public abstract void deactivateThrottling();
 
     /**
+     * Marks operations in the translog from the specified local checkpoint as completed. This is used to restore the state of the local
```

Comment:
yeah, I think it's good to say why it can suddenly go out of sync. That's not normal.